docker/http-to-bucket/spec/app_spec.rb

```diff
@@ -316,4 +316,72 @@ def build_request_url(path = "/", params = {})
     end # context "and an unexpected error occurs"
 
   end # context "when all required parameters are provided"
-end # RSpec.describe
+
+  describe "#clean" do
+    context "when it's given a single layer hash" do
+      it "correctly removes empty hash elements" do
+        object = {foo: {}, bar: "baz"}
+        expectation = {bar: "baz"}
+
+        expect(clean(object)).to eq(expectation)
+      end
+    end
+
+    context "when it's given an array of hashes" do
+      it "correctly removes empty hash elements" do
+        object = [{foo: {}, bar: "baz"}, {foo: "bar", baz: {}}]
+        expectation = [{bar: "baz"}, {foo: "bar"}]
+
+        expect(clean(object)).to eq(expectation)
+      end
+    end
+
+    context "when it's given a multi layered hash" do
+      it "correctly removes empty hash elements" do
+        object = {
+          a: {foo: {}, bar: "baz"},
+          b: {foo: "bar", baz: {}},
+          c: {foo: {bar: "baz", foo: {}}}
+        }
+        expectation = {
+          a: {bar: "baz"},
+          b: {foo: "bar"},
+          c: {foo: {bar: "baz"}}
+        }
+
+        expect(clean(object)).to eq(expectation)
+      end
+    end
+
+    context "when it's given an array of many multi layered hash" do
```

Comment:
This test seems redundant, given that earlier tests prove the array handling and recursion. I might be wrong. Not a big deal.